src/internal/operations/test/exchange_test.go

```diff
@@ -842,3 +845,333 @@ func testExchangeContinuousBackups(suite *ExchangeBackupIntgSuite, toggles contr
 		})
 	}
 }
+
+type ExchangeRestoreIntgSuite struct {
+	tester.Suite
+	its intgTesterSetup
+}
+
+func TestExchangeRestoreIntgSuite(t *testing.T) {
+	suite.Run(t, &ExchangeRestoreIntgSuite{
+		Suite: tester.NewIntegrationSuite(
+			t,
+			[][]string{tconfig.M365AcctCredEnvs, storeTD.AWSStorageCredEnvs}),
+	})
+}
+
+func (suite *ExchangeRestoreIntgSuite) SetupSuite() {
+	suite.its = newIntegrationTesterSetup(suite.T())
+}
+
+func (suite *ExchangeRestoreIntgSuite) TestRestore_Run_exchangeWithAdvancedOptions() {
```

Comment:
I think this test is good for checking broad strokes of behavior but it doesn't ensure that corso takes the proper action on a per item basis. For example, assume no prior knowledge about how corso operated under the hood. In that case, an implementation that that looked at folder names and decided to take the collision policy action would also pass this test

To ensure proper behavior for individual items we should add a test that restores a subset of items in a folder to start with and have the different policies restore a (possibly disjoint) set of the remaining items in the folder

For example, backup a folder with items 1, 2, 3, and 4. Start by restoring item 1. Then, for each policy restore the next as-of-yet unrestored item and the first item (so [1, 2], [1, 3]...). Or if you didn't want to deal with the fact that item 1's ID would change for the replace policy, backup a folder with 6 items and restore 3 to start with and do [1, 4], [2, 5], [3, 6] for the different policies

Reply:
I'm a little hesitant to run with that implementation as it ends up being a lot of lift and complexity for a relatively small gain.  The primary difference from restoring distinct subsets with expanding data would be to ensure that the collision case handling can pivot on a _per-item_ basis, as opposed to a _per-restore_ basis.  That's a level of granularity that's easyto guarantee with a unit test or a mocked integration test.  I can go look and see if we're adequately testing for that sort of complexity at a lower level, and add those tests if they're missing.  As for here, in the operations layer, I think it's acceptable to work with broader strokes, provided we have those granular tests covered elsewhere.

Reply:
were you able to find any more granular tests?

Reply:
Yup, each exchange data category has a per-collision set of tests like [this one](https://github.com/alcionai/corso/blob/main/src/internal/m365/exchange/mail_restore_test.go#L102).  And drive restores have a similar test case [here](https://github.com/alcionai/corso/blob/main/src/internal/m365/onedrive/restore_test.go#L329).  I need to go hook up the new counters to those as assurance.  Otherwise, I think that covers us on the implementation regression in a per-item basis.

Reply:
after some out of band discussion the plan is to get this PR merged along with ID checks for items. After that we can go back and fill out testing a bit more by adding some test cases where only some of the items being restored have the collision policy applied